jboss-eap-config-elytron/added/launch/elytron.sh

```diff
@@ -0,0 +1,130 @@
+# only processes a single environment as the placeholder is not preserved
+
+function prepareEnv() {
+  unset HTTPS_NAME
+  unset HTTPS_PASSWORD
+  unset HTTPS_KEY_PASSWORD
+  unset HTTPS_KEYSTORE_DIR
+  unset HTTPS_KEYSTORE
+  unset HTTPS_KEYSTORE_TYPE
+}
+
+function configure() {
+  configure_https
+  configure_security_domains
+}
+
+function configureEnv() {
+  configure
+}
+
+function configure_https() {
+  local ssl="<!-- No SSL configuration discovered -->"
+  local https_connector="<!-- No HTTPS configuration discovered -->"
+
+  if [ "${CONFIGURE_ELYTRON_SSL}" != "true" ]; then
+    echo "Using PicketBox SSL configuration."
+    return 
+  fi
+
+  if [ -n "${HTTPS_PASSWORD}" -a -n "${HTTPS_KEYSTORE_DIR}" -a -n "${HTTPS_KEYSTORE}" -a -n "${HTTPS_KEYSTORE_TYPE}" ]; then
+
+    if [ -n "${HTTPS_KEY_PASSWORD}" ]; then
+      key_password="<credential-reference clear-text=\"${HTTPS_KEY_PASSWORD}\"/>"
+    fi
+
+    legacy_elytron_tls="<tls>\n\
+        <key-stores>\n\
+            <key-store name=\"LocalhostKeyStore\">\n\
+                <credential-reference clear-text=\"${HTTPS_PASSWORD}\"/>\n\
+                <implementation type=\"${HTTPS_KEYSTORE_TYPE}\"/>\n\
+                <file path=\"${HTTPS_KEYSTORE_DIR}/${HTTPS_KEYSTORE}\"/>\n\
```

Comment:
This is a breaking change, as the meaning of HTTPS_KEYSTORE_DIR has never been a FS path. If anyone provided a FS path as the value for it, that would have broken as shown in CLOUD-2406. So anyone setting it must be providing a config model path name. And setting it is required, so anyone using this must be providing a config model path name.

To be compatible I think we need to do it along the lines I mention in CLOUD-2406. Perhaps we could go a bit beyond that and if HTTPS_KEYSTORE doesn't start with '/' and HTTPS_KEYSTORE_DIR does, then log a WARN and concatenate like this.  (I say log a WARN because passing 2 FS vars and asking us to concatenate is a bit silly if we support using an absolute path for HTTPS_KEYSTORE.)

Reply:
Grrr, but in https.sh, HTTPS_KEYSTORE_DIR *is* an FS path.  Fun, fun!

I don't think that changes what I said above though, as 1) the way https.sh uses it would still work and 2) CONFIGURE_ELYTRON_SSL toggles which is used.

Reply:
@bstansberry I'm not sure I follow when you're saying HTTPS_KEYSTORE_DIR isn't a path, we treat it like one in the templates, it references secrets. like for example: https://github.com/luck3y/jboss-eap-7-openshift-image/blob/eap-cd-dev/templates/eap-cd-https-s2i.json#L530

That said, I need to review your comments on CLOUD-2406, so ignore the above if I'm just being ignorant ;)

Reply:
No, you're not being ignorant; that template usage convinces me that is must work as both a logical config model path name and as a FS path. But regardless of what the template provided as a default, if the user set CONFIGURE_ELYTRON_SSL to true, the only way it would work is if they provided a logical path name as the HTTPS_KEYSTORE_DIR value.

It's possible that no one has actually gone to the effort to ignore the hint the default value in the template gives and instead figured out they need to provide a path name, thus letting elytron.sh work. But checking for a leading '/' to distinguish a FS path from a path name seems simple enough.

Reply:
+1, add the absolute path check seems easy enough.

Reply:
I would say it's possible no one actually tried to change the default value of `HTTPS_KEYSTORE_DIR`, which contains leading '/' and is set to `/etc/eap-secret-volume`. It's not exposed between the parameters in the template, so this could be hit only if someone tried to deploy the image directly (without the template).

Reply:
You could also use a template and the -e param to oc new-app.  I won't try and argue this is likely though.